packages/dev/core/src/Materials/Node/Blocks/PBR/pbrMetallicRoughnessBlock.ts

```diff
@@ -70,9 +70,9 @@ export class PBRMetallicRoughnessBlock extends NodeMaterialBlock {
     private static _OnGenerateOnlyFragmentCodeChanged(block: NodeMaterialBlock, _propertyName: string): boolean {
         const that = block as PBRMetallicRoughnessBlock;
 
-        if (that.worldPosition.isConnected) {
+        if (that.worldPosition.isConnected || that.worldNormal.isConnected) {

```

Comment:
@deltakosh - just leaving the comment here to wait for you to review it. Please resolve this when done.